packages/core/src/types.ts

```diff
@@ -1990,6 +1996,10 @@ export interface ActorRef<
       emitted: TEmitted & (TType extends '*' ? unknown : { type: TType })
     ) => void
   ) => Subscription;
+  select<TSelected, TSnapshot>(
```

Comment:
`TSnapshot` is already available in the scope here (it's an existing type parameter of `ActorRef`), so u dont need it here - only `TSelected` is needed

Reply:
I get compilation error when I remove TSnapshot and then run `pnpm typecheck`, see [GitHub Actions](https://github.com/statelyai/xstate/actions/runs/15810540587/job/44561254051?pr=5299#step:4:46). Not sure on how to resolve it. Reverting to use TSnapshot for now.

I looked into adding a generic for TSnapshot on Actor, se below. But that ended up in a lot of changes not really in the scope of this PR. @Andarist any suggestions?

```typescript
export class Actor<TLogic extends AnyActorLogic, TSnapshot extends SnapshotFrom<TLogic>>
  implements
    ActorRef<SnapshotFrom<TLogic>, EventFromLogic<TLogic>, EmittedFrom<TLogic>>
{
  ....
}
```

Reply:
@Andarist Any idea how to resolve this without a major refactor? Is it required in order to merge the PR?